pkg/provider/gcp/secretmanager/workload_identity_federation.go

```diff
@@ -295,7 +295,7 @@ func (w *workloadIdentityFederation) updateExternalAccountConfigWithSubjectToken
 		ns = *w.config.ServiceAccountRef.Namespace
 	}
 	config.SubjectTokenSupplier = &k8sSATokenReader{
-		audience:         config.Audience,
+		audience:         w.config.Audience,
```

Comment:
Nice catch. I assume there were no tests for this? :D

Reply:
Yeah, the manual e2e tests would have caught, not sure when this typo happened :)